package.json

```diff
@@ -18,19 +18,20 @@
   ],
   "exports": {
     ".": {
+      "types": "./types/index.d.ts",
       "import": {
         "node": "./dist/vue.runtime.mjs",
         "default": "./dist/vue.runtime.esm.js"
       },
-      "require": "./dist/vue.runtime.common.js",
-      "types": "./types/index.d.ts"
+      "require": "./dist/vue.runtime.common.js"
     },
     "./compiler-sfc": {
+      "types": "./compiler-sfc/index.d.ts",
       "import": "./compiler-sfc/index.mjs",
       "require": "./compiler-sfc/index.js"
     },
     "./dist/*": "./dist/*",
-    "./types/*": "./types/*",
+    "./types/*": ["./types/*.d.ts", "./types/*"],
```

Comment:
To help clarify the conversion from import name to file lookups, I tried to follow the pseudo-code from the documentation here: https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports and list out what would happen in the the following scenarios:

Before this change:
```
"vue/types/options" => ["./types/options"] // fails to find types
"vue/types/options.js" => ["./types/options.ts", "./types/options.d.ts", "./types/options.js"] // finds on second try
"vue/types/options.d.ts" => ["./types/options.d.ts"] // finds on first try
```

After this change:
```
"vue/types/options" => ["./types/options.d.ts", "./types/options"] // finds on first try
"vue/types/options.js" => ["./types/options.js.d.ts", "./types/options.ts", "./types/options.d.ts", "./types/options.js"] // finds on third try
"vue/types/options.d.ts" => ["./types/options.d.ts.d.ts", "./types/options.d.ts"] // finds on second try
```

Since the most common case is to write these imports without the extensions, I thought it would be best to write this as `["./types/*.d.ts", "./types/*"]`. This makes the common case more direct to resolve.